gateway_pp_test.go

```diff
@@ -0,0 +1,228 @@
+package infrared
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/haveachin/infrared/protocol"
+	"github.com/haveachin/infrared/protocol/handshaking"
+	"github.com/pires/go-proxyproto"
+)
+
+// Checking or proxy is working by checking or the remoteAddr's port is different from the port the client is using
+func getIpFromAddr(addr net.Addr) string {
+	return strings.Split(addr.String(), ":")[0]
+}
+
+func createConnWithFakeIP(gatewayAddr string) (Conn, error) {
+	dialer := &net.Dialer{
+		LocalAddr: &net.TCPAddr{
+			IP:   net.ParseIP("127.0.10.1"),
+			Port: 0,
+		},
+	}
+	netConn, _ := dialer.Dial("tcp", gatewayAddr)
+	return wrapConn(netConn), nil
+}
+
+func TestProxyProtocolOn(t *testing.T) {
+	serverAddr := "infrared.gateway"
+	listenAddr := ":26572"
+	gatewayPort := 25572
+	gatewayAddr := fmt.Sprintf(":%d", gatewayPort)
+
+	cDial := make(chan error)
+	cListen := make(chan error)
+	cGateway := make(chan error)
+	cResult := make(chan bool)
+	cInfo := make(chan string)
```

Comment:
Some of these names seem ambiguous to me. What does the 'c' stand for?

Reply:
'c' stands for channel, I thought it would be nicer for each goroutine to have its own channel to report error it encounters. 

cInfo is meant for communication between goroutines which is needed to determine or the testcase failed or passed.  Haven't found a better name for it yet.